memo/anchor_program/memo/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+use anchor_lang::prelude::*;
+
+declare_id!("4ARKJpztrkSJXJQ8ii9pr6NjJdwKrC6M3jA8LG21tvjG");
+
+#[program]
+pub mod memo {
+    use super::*;
+    use std::str;
+    pub fn build_memo(ctx: Context<BuildMemo>,input:Vec<u8>) -> Result<()> {
```

Comment:
- Changed the function name to ```log_memo```
- Anchor deserialize function won't allow the input to be ```&[u8]```, tried a lot of different ways to make it happen but well Anchor complains a lot in this particular scenario. 
Error: 
```the function or associated item `deserialize` exists for struct `instruction::LogMemo`, but its trait bounds were not satisfied```

Reply:
Ah, this might be the place for zero-copy deserialization! Here's an example for you to follow: https://github.com/project-serum/anchor/blob/master/tests/zero-copy/programs/zero-copy/src/lib.rs